crates/bevy_ecs/src/query/mod.rs

```diff
@@ -51,6 +51,20 @@ mod tests {
         assert_eq!(values, vec![&B(3)]);
     }
 
+    #[test]
+    fn query_filtered_len() {
+        let mut world = World::new();
+        world.spawn().insert_bundle((A(1), B(1)));
+        world.spawn().insert_bundle((A(2),));
+        world.spawn().insert_bundle((A(3),));
+
+        let mut values = world.query_filtered::<&A, With<B>>();
+        assert_eq!(values.iter(&world).len(), 1);
+
+        let mut values = world.query_filtered::<&A, Without<B>>();
+        assert_eq!(values.iter(&world).len(), 2);
```

Comment:
I'd be more confident if those tests also made sure more complex filters like `Or<…>` and `(With<X>, Without<Y>, Or<…>)` worked.

Reply:
I agree that I should have tested more complex filters.  I'll get back on that.